http/base/src/main/java/org/apache/felix/http/base/internal/dispatch/ServletRequestWrapper.java

```diff
@@ -411,143 +381,17 @@ public boolean isAsyncSupported()
-                    return ServletRequestWrapper.this.getCharacterEncoding();
-                }
-            };
-        }
-        return multipartContext;
-    }
-
-    private Collection<PartImpl> checkMultipart() throws IOException, ServletException {
-        if ( parts == null ) {
-            final RequestContext multipartContext = getMultipartContext();
-            if ( multipartContext != null && FileUploadBase.isMultipartContent(multipartContext) ) {
-                if ( this.multipartConfig == null) {
-                    throw new IllegalStateException("Multipart not enabled for servlet.");
-                }
-
-                if ( System.getSecurityManager() == null ) {
-                    handleMultipart(multipartContext);
-                } else {
-                    final AccessControlContext ctx = bundleForSecurityCheck.adapt(AccessControlContext.class);
-                    final IOException ioe = AccessController.doPrivileged(new PrivilegedAction<IOException>() {
-
-                        @Override
-                        public IOException run() {
-                            try {
-                                handleMultipart(multipartContext);
-                            } catch ( final IOException ioe) {
-                                return ioe;
-                            }
-                            return null;
-                        }
-                    }, ctx);
-                    if ( ioe != null ) {
-                        throw ioe;
-                    }
-                }
-
-            } else {
-                throw new ServletException("Not a multipart request");
-            }
-        }
-        return parts;
-    }
-
-    private void handleMultipart(final RequestContext multipartContext) throws IOException {
-        // Create a new file upload handler
-        final FileUpload upload = new FileUpload();
-        upload.setSizeMax(this.multipartConfig.multipartMaxRequestSize);
-        upload.setFileSizeMax(this.multipartConfig.multipartMaxFileSize);
-        upload.setFileItemFactory(new DiskFileItemFactory(this.multipartConfig.multipartThreshold,
-                new File(this.multipartConfig.multipartLocation)));
-        upload.setFileCountMax(this.multipartConfig.multipartMaxFileCount);
-        // Parse the request
-        List<FileItem> items = null;
-        try {
-            items = upload.parseRequest(multipartContext);
-        } catch (final FileUploadException fue) {
-            throw new IOException("Error parsing multipart request", fue);
-        }
-        this.parts = new ArrayList<>();
-        for(final FileItem item : items) {
-            this.parts.add(new PartImpl(item));
-        }
-    }
-
     @Override
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public Collection<Part> getParts() throws IOException, ServletException {
-        return (Collection)checkMultipart();
-
+        return null;
```

Comment:
According to the servlet spec, this method either throws a ServletException if the request is not a multipart request or always returns a collection. It never returns null. Throwing an exception seems to be the right thing

Reply:
Thank you!
I now throw an ServletException.